### PR TITLE
Attach keyboard shortcuts to features

### PR DESCRIPTION
**Pull request by yanas**
Shortcuts are now registered dynamically through the toolbar button they represent. Only enabled buttons would register a shortcut. Thus only the registered shortcuts will appear in the help menu. I have also added an "unregisterShortcut" method, which isn't currently used. So an open question is, do we need to unregister a shortcut at some point, keeping in mind that we cannot currently enable/disable a feature (toolbar button) during runtime.

### CODE REVIEW

File: modules/UI/toolbars/Toolbar.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 /* jshint -W101 */
 import UIUtil from '../util/UIUtil';
 import UIEvents from '../../../service/UI/UIEvents';
+import KeyboardShortcut from '../../keyboardshortcut/keyboardshortcut';
```
Comment (hristoterezov) [inline modules/UI/toolbars/Toolbar.js]: Since KeyboardShortcut is separate module isn't it better to be attached to the global APP object like other modules. That way you should access it with `APP. KeyboardShortcut` instead of `import`. WDYT?

File: modules/keyboardshortcut/keyboardshortcut.js [hunk 2]
```diff
@@ -1,91 +1,64 @@
...
+        JitsiMeetJS.analytics.sendEvent("shortcut.raisedhand.toggled");
+        APP.conference.maybeToggleRaisedHand();
+    }, "keyboardShortcuts.raiseHand");
+
+    KeyboardShortcut.registerShortcut("T", null, function() {
+        JitsiMeetJS.analytics.sendEvent("shortcut.talk.clicked");
+        APP.conference.muteAudio(true);
+    }, "keyboardShortcuts.pushToTalk");
+
+    KeyboardShortcut.registerShortcut("F", 'filmstripPopover', function() {
```
Comment (hristoterezov) [inline modules/keyboardshortcut/keyboardshortcut.js]: I think it's going to be awesome if we register this shortcut from the BottomToolbar.

File: modules/keyboardshortcut/keyboardshortcut.js [hunk 3]
```diff
@@ -112,20 +85,56 @@ var KeyboardShortcut = {
...
+            this._addShortcutToHelp(shortcutChar, helpDescription);
+    },
+
+    /**
+     * Unregisters a shortcut.
+     *
+     * @param shortcutChar unregisters the given shortcut, which means it will
+     * no longer be usable
+     */
+    unregisterShortcut: function(shortcutChar) {
```
Comment (hristoterezov) [inline modules/keyboardshortcut/keyboardshortcut.js]: Shouldn't we remove the shortcut from the help html when this method is called?

File: modules/keyboardshortcut/keyboardshortcut.js [hunk 4]
```diff
@@ -1,91 +1,64 @@
...
+    }, "keyboardShortcuts.pushToTalk");
+
+    KeyboardShortcut.registerShortcut("F", 'filmstripPopover', function() {
+        JitsiMeetJS.analytics.sendEvent("shortcut.film.toggled");
+        APP.UI.toggleFilmStrip();
+    }, "keyboardShortcuts.toggleFilmstrip");
+
+    // Focus keys are directly implemented below.
+    KeyboardShortcut._addShortcutToHelp("0", "keyboardShortcuts.focusLocal");
+    KeyboardShortcut._addShortcutToHelp("1-9", "keyboardShortcuts.focusRemote");
```
Comment (hristoterezov) [inline modules/keyboardshortcut/keyboardshortcut.js]: How are these shortcuts with the digits working? Maybe we should somehow register them from SmallVideo if it is possible? WDYT?

We can do this trough another PR if it is possible and if you like the idea.

**Comment by hristoterezov** [comment]
The idea of registering the shortcuts dynamically from different modules is awesome!

File: modules/keyboardshortcut/keyboardshortcut.js (hunk 4 above)
Comment (yanas) [inline modules/keyboardshortcut/keyboardshortcut.js]: Those were handled directly in the key event handler and I would probably need to refactor the code in a separate PR. I'll add a comment there. I like the idea very much indeed.
